graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/FederatedSchemaGeneratorHooks.kt

```diff
@@ -72,13 +72,17 @@ open class FederatedSchemaGeneratorHooks(private val federatedTypeRegistry: Fede
     override fun willBuildSchema(builder: GraphQLSchema.Builder): GraphQLSchema.Builder {
         val originalSchema = builder.build()
         val originalQuery = originalSchema.queryType
-
         val federatedCodeRegistry = GraphQLCodeRegistry.newCodeRegistry(originalSchema.codeRegistry)
-        val federatedSchema = GraphQLSchema.newSchema(originalSchema)
+        val federatedSchemaBuilder = GraphQLSchema.newSchema(originalSchema)
         val federatedQuery = GraphQLObjectType.newObject(originalQuery)
             .field(SERVICE_FIELD_DEFINITION)
             .withDirective(extendsDirectiveType)
 
+        // Add the @extends directive definition if it is not defined as we are using it on the Query object
+        if (originalSchema.getDirective(EXTENDS_DIRECTIVE_NAME) == null) {
+            federatedSchemaBuilder.additionalDirective(extendsDirectiveType)
```

Comment:
should we add all federated directives here?

Reply:
The only other things we do with the `federatedSchemaBuilder` is to add the `_Any` scalar and update the `Query` object. So all the other directive definitions will be their if they are used, otherwise if they are not there it means they are not used by the schema do we don't need to add it

Reply:
I was wondering whether all federated directives should always be present (i.e. similar behavior to default `include`, `skip` and `deprecated`).

Reply:
I suppose we could. It looks like the apollo-federation lib adds them to the schema

See this running example of federation and the schema tab: https://pw678w138q.sse.codesandbox.io/graphql